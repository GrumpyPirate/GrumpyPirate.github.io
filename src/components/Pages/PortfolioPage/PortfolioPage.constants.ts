import Color from 'color';
import styled from 'styled-components';

import {
  animDur,
  forceAspectRatio,
  gridGutterWidthBase,
  gutterWidths,
  listReset,
  mediaQueries,
  palette,
  rem,
} from 'styles';

// Styled components
export const List = styled.ul`
  ${listReset}
  margin: 0 ${rem(gutterWidths.xs * -0.25)};
  padding-top: ${rem(gutterWidths.xs * 0.5)};

  @media ${mediaQueries.md} {
    display: flex;
    flex-wrap: wrap;
    margin: 0 ${rem(gutterWidths.md * -0.25)};
    padding-top: ${rem(gutterWidths.md * 0.5)};
  }
`;

export const ListItemImage = styled.figure`
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  right: 0;
  top: 0;
  transition: transform ${animDur}s ease;

  img {
    bottom: 0;
    height: 100%;
    left: 0;
    object-fit: cover;
    position: absolute;
    right: 0;
    top: 0;
    width: 100%;
  }
`;

export const ListItemCopy = styled.div`
  background-image: linear-gradient(
    to top,
    ${Color(palette.themeDarkShades).alpha(0.8).string()},
    ${Color(palette.themeDarkShades).alpha(0.4).string()} 40%,
    ${Color(palette.themeDarkShades).alpha(0).string()}
  );
  color: ${palette.white};
  padding: 15vmax ${rem(gridGutterWidthBase / 2)} ${rem(gridGutterWidthBase / 2)};
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: 1;
`;

export const ListItemDescription = styled.p`
  margin-bottom: 0;
`;

export const ListItem = styled.li`
  display: block;
  margin-right: ${rem(gutterWidths.xs / 4)};
  margin-bottom: ${rem(gutterWidths.xs / 2)};
  margin-left: ${rem(gutterWidths.xs / 4)};
  text-align: left;

  @media ${mediaQueries.sm} {
    flex-basis: calc(50% - ${rem(gutterWidths.xs / 2)});
    margin-right: ${rem(gutterWidths.xs / 4)};
    margin-bottom: ${rem(gutterWidths.xs / 2)};
    margin-left: ${rem(gutterWidths.xs / 4)};
  }

  @media ${mediaQueries.md} {
    flex-basis: calc(50% - ${rem(gutterWidths.md / 2)});
    margin-right: ${rem(gutterWidths.md / 4)};
    margin-bottom: ${rem(gutterWidths.md / 2)};
    margin-left: ${rem(gutterWidths.md / 4)};
  }

  @media ${mediaQueries.lg} {
    border-bottom-width: ${rem(gutterWidths.md / 4)};
  }

  &:hover,
  &:focus {
    ${ListItemImage} {
      transform: scale(1.0625);
    }
  }

  a {
    ${forceAspectRatio(16, 9)}
    position: relative;
    display: flex;
    flex: 1 1 calc(100% - ${rem(gutterWidths.xs / 2)});
    flex-direction: column;
    justify-content: flex-end;
    overflow: hidden;
    border: ${rem(gutterWidths.xs / 4)} solid ${palette.themeAccentDark};
    border-style: none none solid;
    background-color: ${palette.greyDark};
  }
`;
