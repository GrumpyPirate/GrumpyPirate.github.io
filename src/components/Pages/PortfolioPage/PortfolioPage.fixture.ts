import { PortfolioItemFormatted } from 'types/common';

const portfolioItems: PortfolioItemFormatted[] = [
  {
    createdAt: '6/27/2045',
    id: '93fffc74-34a4-49da-a995-eed94dfdc9bc',
    slug: 'scrupulum-misphrase',
    title: 'Scrupulum Misphrase',
    descriptionShort: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo velit omnis.',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati.\n\nLorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati.',
    url:
      'https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform',
    tech: [
      {
        icon: 'storybook',
        name: 'youngling',
      },
    ],
    supportingImageSrc: 'http://autotomic.com/envious/israelitish?a=rhonchal&b=tetrahydro#piehouse',
    headerImgSrc: 'https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience',
    previews: {
      desktop: 'http://hyperterrestrial.com/unconspiringness/uncreeping?a=pasterer&b=hemen#timalia',
      tablet: 'https://subtilism.com/dermatoscopy/nombril?a=phlebolithiasis&b=blancher#downthrow',
      mobile: 'https://quadrupedan.com/inroader/ohmmeter?a=proseuche&b=provisorily#coriander',
    },
  },
  {
    createdAt: '6/27/2045',
    id: 'ecd3cfe6-c880-4c7e-a15c-f13e27cf5c69',
    slug: 'scrupulum-misphrase',
    title: 'Scrupulum Misphrase',
    descriptionShort: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo velit omnis.',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati.\n\nLorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati.',
    url:
      'https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform',
    tech: [
      {
        icon: 'storybook',
        name: 'youngling',
      },
    ],
    supportingImageSrc: 'http://autotomic.com/envious/israelitish?a=rhonchal&b=tetrahydro#piehouse',
    headerImgSrc: 'https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience',
    previews: {
      desktop: 'http://hyperterrestrial.com/unconspiringness/uncreeping?a=pasterer&b=hemen#timalia',
      tablet: 'https://subtilism.com/dermatoscopy/nombril?a=phlebolithiasis&b=blancher#downthrow',
      mobile: 'https://quadrupedan.com/inroader/ohmmeter?a=proseuche&b=provisorily#coriander',
    },
  },
  {
    createdAt: '6/27/2045',
    id: '09d58d40-16f9-451c-adec-04cfc9d5655d',
    slug: 'scrupulum-misphrase',
    title: 'Scrupulum Misphrase',
    descriptionShort: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo velit omnis.',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati.\n\nLorem ipsum dolor sit amet, consectetur adipisicing elit. Modi deserunt veritatis dolore sit ea, illum quae omnis placeat, natus est sapiente atque eaque obcaecati.',
    url:
      'https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform',
    tech: [
      {
        icon: 'storybook',
        name: 'youngling',
      },
    ],
    supportingImageSrc: 'http://autotomic.com/envious/israelitish?a=rhonchal&b=tetrahydro#piehouse',
    headerImgSrc: 'https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience',
    previews: {
      desktop: 'http://hyperterrestrial.com/unconspiringness/uncreeping?a=pasterer&b=hemen#timalia',
      tablet: 'https://subtilism.com/dermatoscopy/nombril?a=phlebolithiasis&b=blancher#downthrow',
      mobile: 'https://quadrupedan.com/inroader/ohmmeter?a=proseuche&b=provisorily#coriander',
    },
  },
];

export default portfolioItems;
