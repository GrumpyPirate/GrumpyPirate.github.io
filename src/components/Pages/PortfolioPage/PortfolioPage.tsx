import { useQuery } from '@apollo/client';
import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import Page from 'components/Page/Page';
import PageContent from 'components/PageContent/PageContent';
import PageHeader from 'components/PageHeader/PageHeader';
import PageHeaderSubtitle from 'components/PageHeaderSubtitle/PageHeaderSubtitle';
import PortfolioList from 'components/PortfolioList/PortfolioList';
import Spinner from 'components/Spinner/Spinner';
import { GET_PORTFOLIO_ITEM_LIST, GetPortfolioItemsResponse } from 'queries';
import { PropsWithClassName } from 'types/common';

const PortfolioPage: FunctionComponent<PropsWithClassName> = function ({ className }) {
  const { loading } = useQuery<GetPortfolioItemsResponse>(GET_PORTFOLIO_ITEM_LIST);

  return (
    <Page className={className}>
      <PageHeader title="Portfolio">
        <PageHeaderSubtitle>Projects I&apos;ve worked on</PageHeaderSubtitle>
      </PageHeader>

      <PageContent>
        {loading && <Spinner />}
        <PortfolioList />
      </PageContent>
    </Page>
  );
};

export default styled(PortfolioPage)`
  ${PageContent} {
    padding-top: 0;
  }
`;
