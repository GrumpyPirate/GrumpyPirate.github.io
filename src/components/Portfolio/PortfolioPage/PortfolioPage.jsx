// React
import React, { PureComponent } from 'react'

// Vendor
import marked from 'marked'

// Styles
// import './PortfolioPage.scss'

// Components
import PortfolioPageHeader from 'components/Portfolio/PortfolioPageHeader/PortfolioPageHeader.jsx'
import PortfolioPageContent from 'components/Portfolio/PortfolioPageContent/PortfolioPageContent.jsx'
import PortfolioCarousel from 'components/Portfolio/PortfolioCarousel/PortfolioCarousel.jsx'
import PortfolioDeviceLineup from 'components/Portfolio/PortfolioDeviceLineup/PortfolioDeviceLineup.jsx'

// Services
import ContentService from 'services/ContentService'

class PortfolioPage extends PureComponent {
  constructor (props) {
    super(props)

    this.state = {
      contentService: new ContentService(),
      portfolioItem: null,
      portfolioItemSlug: this.props.match.params.portfolioItemSlug
    }
  }

  getDescription (markdownDescription) {
    const markedOptions = {
      sanitize: true
    }

    return marked(this.state.portfolioItem.description, markedOptions)
  }

  componentWillMount () {
    this.props.emitter.emit('startLoading')

    this.state.contentService
      .getSinglePortfolioItem(this.state.portfolioItemSlug)
      .then((item) => {
        this.setState({
          portfolioItem: item
        })

        this.props.emitter.emit('stopLoading')
      })
      .catch((e) => {
        console.error(e)
        this.props.emitter.emit('stopLoading')
      })
  } // /componentWillMount

  render () {
    return (
      <div className="pf-page">
        {this.state.portfolioItem &&
          <div>
            <PortfolioPageHeader bgImage={this.state.portfolioItem.headerImgSrc} title={this.state.portfolioItem.title} tech={this.state.portfolioItem.tech}/>

            {/* Content */}
            <PortfolioPageContent>
              <div className="container-fluid">
                <div className="row justify-content-center align-items-center">
                  <div className="col-12 col-sm-10">
                    {this.state.portfolioItem.supportingImageSrc
                      ? (
                        <div className="row align-items-lg-center">
                          <div className="col-12 col-md" dangerouslySetInnerHTML={{__html: this.getDescription(this.state.portfolioItem.description)}}></div>

                          <div className="col-12 col-md-6 col-xl-5">
                            {/* Supporting image 1 */}
                            <hr className="mt-1 hidden-md-up"/>
                            <figure className="mb-0">
                              <img src={this.state.portfolioItem.supportingImageSrc} alt="" className="w-100" />
                            </figure>
                          </div>
                        </div>
                      )
                      : (
                        <div>
                          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quae enim quisquam exercitationem doloremque porro veritatis repellat ea, mollitia maxime, ullam aliquam! Deleniti, blanditiis nostrum.
                          </p>
                          <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi, deserunt. Iste reiciendis magnam temporibus quidem, quis hic libero molestiae. Cum facere, recusandae reiciendis corrupti ipsa placeat quas explicabo rem debitis quod magnam sequi. Minus pariatur totam, maxime deleniti a ullam cum rem praesentium, minima reiciendis, ad quaerat repudiandae.
                          </p>
                        </div>
                      )
                    }

                    <hr/>

                    {/* Device Carousel, for mobile (sm-) only */}
                    <div className="hidden-md-up">
                      <PortfolioCarousel
                        desktop={this.state.portfolioItem.previews.desktop}
                        tablet={this.state.portfolioItem.previews.tablet}
                        mobile={this.state.portfolioItem.previews.mobile} />
                    </div>

                    {/* Device Lineup, for tablet up */}
                    <div className="hidden-sm-down">
                      <PortfolioDeviceLineup
                        desktop={this.state.portfolioItem.previews.desktop}
                        tablet={this.state.portfolioItem.previews.tablet}
                        mobile={this.state.portfolioItem.previews.mobile} />
                    </div>

                    <hr/>

                    <div className="text-center mt-2 mt-md-4">
                      <a href={this.state.portfolioItem.url} className="btn btn-secondary" target="_blank">Visit {this.state.portfolioItem.title}</a>
                    </div>
                  </div>
                </div>
              </div>
            </PortfolioPageContent>
          </div>
        }
      </div>
    )
  } // /render()
} // /class PortfolioPage extends Component

export default PortfolioPage
