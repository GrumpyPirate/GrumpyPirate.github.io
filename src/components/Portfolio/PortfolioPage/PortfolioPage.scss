@import 'core';

.pf-page {
  &__supporting-image {
    border-radius: .25rem;
  } // /&__supporting-image
} // /.pf-page
