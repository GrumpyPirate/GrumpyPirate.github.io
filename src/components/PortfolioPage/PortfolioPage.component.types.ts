import { RouteComponentProps } from 'react-router';
import { PortfolioItemFormatted } from 'types/common';

export interface PortfolioPageProps extends RouteComponentProps {
  hasFetched: boolean;
  portfolioItem: PortfolioItemFormatted;
  requestPortfolioItems: () => void;
}
