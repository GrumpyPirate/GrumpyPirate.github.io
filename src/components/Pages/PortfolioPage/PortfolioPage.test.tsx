import { render } from '@testing-library/react';
import React from 'react';

import { initialRootState, RootState } from 'store';
import { withMockRouter, withMockStore } from 'utils/testing';

import PortfolioPage from './PortfolioPage';
import portfolioItems from './PortfolioPage.fixture';

describe('Components', () => {
  describe('Portfolio', () => {
    const mockState: RootState = {
      ...initialRootState,
      portfolio: {
        ...initialRootState.portfolio,
        portfolioItems,
      },
    };

    it('should render as expected, without crashing', () => {
      const { container } = render(withMockRouter(withMockStore(<PortfolioPage />, mockState)));

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
