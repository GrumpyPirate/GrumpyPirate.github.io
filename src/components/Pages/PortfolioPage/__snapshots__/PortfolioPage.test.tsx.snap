// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Portfolio should render as expected, without crashing 1`] = `
.c0 {
  background-color: hsl(0,0%,95%);
  min-height: 100vh;
  position: relative;
}

.c10 {
  position: relative;
  padding-bottom: 1.5rem;
  padding-top: 1.5rem;
  text-align: center;
  min-height: 100vh;
}

.c8 {
  position: relative;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  max-width: 100%;
  min-height: 1px;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c3 {
  width: 100%;
  padding-right: 0.75rem;
  padding-left: 0.75rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  margin-right: -0.75rem;
  margin-left: -0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  background-color: #302445;
  border-bottom: solid 0.375rem #dc8381;
  text-align: center;
  text-transform: uppercase;
}

.c4 {
  color: #dc8381;
  padding: 1rem 0;
  position: relative;
}

.c5 {
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 2.475rem;
  font-size: 12vmin;
  font-weight: 300;
  margin: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c5::first-letter {
  margin-left: 0.25em;
}

.c6 {
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
  color: rgba(242,241,243,0.5);
  margin: 0;
  padding-bottom: 1rem;
  text-transform: uppercase;
}

.c6 a {
  color: #dc8381;
}

.c6 a:hover,
.c6 a:focus {
  color: hsl(1.3000000000000114,56.5%,75.3%);
}

.c6 a:active {
  color: hsl(1.3000000000000114,56.5%,61.6%);
}

.c18 {
  opacity: 0;
  visibility: hidden;
  -webkit-transition-property: visibility,opacity;
  transition-property: visibility,opacity;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-timing-function: ease;
  transition-timing-function: ease;
}

.c20 {
  font-family: 'Barlow',sans-serif;
  font-weight: 300;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  line-height: 1.2;
  margin: 0 0 1rem;
  text-transform: uppercase;
  font-size: 1rem;
}

.c17 {
  bottom: 0;
  left: 0;
  margin: 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: -webkit-transform 0.15s ease;
  -webkit-transition: transform 0.15s ease;
  transition: transform 0.15s ease;
}

.c17 img {
  bottom: 0;
  height: 100%;
  left: 0;
  object-fit: cover;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

.c19 {
  background-image: linear-gradient( to top, rgba(48,36,69,0.8), rgba(48,36,69,0.4) 40%, rgba(48,36,69,0) );
  color: #fff;
  padding: 15vmax 0.9375rem 0.9375rem;
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
  z-index: 1;
}

.c21 {
  margin-bottom: 0;
}

.c16 {
  display: block;
}

.c16 a {
  height: 0;
  padding-bottom: 56.25%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 calc(100% - 0.75rem);
  -ms-flex: 1 1 calc(100% - 0.75rem);
  flex: 1 1 calc(100% - 0.75rem);
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  text-align: left;
  overflow: hidden;
  font-weight: 300;
  border: 0.375rem solid #b54259;
  border-style: none none solid;
  border-radius: 0.25rem;
  background-color: #333;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
}

.c16 a:hover,
.c16 a:focus {
  -webkit-transform: translateY(-0.25rem);
  -ms-transform: translateY(-0.25rem);
  transform: translateY(-0.25rem);
  -webkit-transition-duration: 0s;
  transition-duration: 0s;
}

.c11 {
  display: grid;
  grid-gap: 0.75rem;
  padding: 0.75rem 0.75rem 0;
  text-align: left;
}

.c13 {
  margin-right: 0.5rem;
}

.c14 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c12 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 2;
}

.c15 {
  list-style: none;
  margin: 0;
  padding: 0;
  display: grid;
  grid-gap: 0.75rem;
  padding: 0.75rem 0.75rem 0;
}

.c1 .c9 {
  padding-top: 0;
}

@media (min-width:48em) {
  .c10 {
    padding-bottom: 3rem;
    padding-top: 3rem;
  }
}

@media (min-width:48em) {
  .c8 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:30em) {
  .c8 {
    -webkit-flex-basis: 83.33333333333334%;
    -ms-flex-preferred-size: 83.33333333333334%;
    flex-basis: 83.33333333333334%;
    max-width: 83.33333333333334%;
  }
}

@media (min-width:48em) {
  .c8 {
    -webkit-flex-basis: 66.66666666666666%;
    -ms-flex-preferred-size: 66.66666666666666%;
    flex-basis: 66.66666666666666%;
    max-width: 66.66666666666666%;
  }
}

@media (min-width:48em) {
  .c3 {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
}

@media (min-width:48em) {
  .c7 {
    margin-right: -0.9375rem;
    margin-left: -0.9375rem;
  }
}

@media (min-width:75em) {
  .c2 {
    border-bottom-width: 0.5rem;
  }
}

@media (min-width:48em) {
  .c4 {
    padding-top: 2rem;
  }
}

@media (min-width:48em) {
  .c5 {
    font-size: 4.5rem;
  }
}

@media (min-width:48em) {
  .c6 {
    font-size: 1.5rem;
  }
}

@media (min-width:48em) {
  .c6 {
    padding-bottom: 2rem;
  }
}

@media (min-width:48em) {
  .c20 {
    font-size: 1rem;
  }
}

@media (min-width:48em) {
  .c16 {
    opacity: 0;
    -webkit-animation: hpzHus 0.15s ease 0s 1 forwards;
    animation: hpzHus 0.15s ease 0s 1 forwards;
  }
}

@media (min-width:75em) {
  .c16 a {
    border-bottom-width: 0.46875rem;
  }
}

@media (min-width:48em) {
  .c11 {
    grid-template-columns: 1fr 1fr;
    grid-gap: 0.9375rem;
    padding: 0.9375rem 0.9375rem 0;
  }
}

@media (min-width:75em) {
  .c11 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media (min-width:48em) {
  .c12 {
    grid-column-start: 2;
  }
}

@media (min-width:75em) {
  .c12 {
    grid-column-start: 3;
  }
}

@media (min-width:48em) {
  .c15 {
    grid-template-columns: 1fr 1fr;
    grid-gap: 0.9375rem;
    padding: 0.9375rem 0.9375rem 0;
  }
}

@media (min-width:75em) {
  .c15 {
    grid-template-columns: 1fr 1fr 1fr;
  }
}

@media (min-width:48em) {

}

<div
  class="c0 c1"
>
  <header
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class=""
      >
        <div
          class="c4"
        >
          <h1
            class="c5"
          >
            Portfolio
          </h1>
        </div>
        <h2
          class="c6"
        >
          <div
            class="c3"
          >
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                Projects I've worked on
              </div>
            </div>
          </div>
        </h2>
      </div>
    </div>
  </header>
  <div
    class="c9 c10"
  >
    <div>
      <div
        class="c11"
      >
        <div
          class="c12"
        >
          <label
            class="c13"
            for="sort-select"
          >
            Sort by:
          </label>
          <div
            class="c14"
          >
            <div
              class=" css-2b097c-container"
            >
              <div
                class=" css-p93uir-control"
              >
                <div
                  class=" css-g1d714-ValueContainer"
                >
                  <div
                    class=" css-1uccc91-singleValue"
                  >
                    Date (Newest first)
                  </div>
                  <input
                    aria-autocomplete="list"
                    class="css-62g3xt-dummyInput"
                    id="sort-select"
                    readonly=""
                    tabindex="0"
                    value=""
                  />
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <ul
        class="c15"
      >
        <li
          class="c16"
          data-testid="portfolio-page__list__item"
          style="animation-delay: 0s;"
        >
          <a
            href="/portfolio/scrupulum-misphrase"
            role="listitem"
          >
            <figure
              class="c17"
            >
              <picture>
                <source
                  sizes="360px"
                  srcset="https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=webp&w=360 360w, https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=webp&w=720 720w"
                  type="image/webp"
                />
                <source
                  sizes="360px"
                  srcset="https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=jpg&w=360 360w, https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=jpg&w=720 720w"
                  type="image/jpeg"
                />
                <source
                  sizes="600px"
                  srcset="https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=webp&w=600 600w, https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=webp&w=1200 1200w"
                  type="image/webp"
                />
                <source
                  sizes="600px"
                  srcset="https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=jpg&w=600 600w, https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience?fm=jpg&w=1200 1200w"
                  type="image/jpeg"
                />
                <img
                  alt=""
                  class="c18"
                  data-testid="loadable-image"
                  loading="lazy"
                  src="https://devotement.com/anticorn/pajahuello?a=rhynchophoran&b=toyer#prescience"
                />
              </picture>
            </figure>
            <div
              class="c19"
            >
              <h3
                class="c20"
              >
                Scrupulum Misphrase
              </h3>
              <p
                class="c21"
              >
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo velit omnis.
              </p>
            </div>
          </a>
        </li>
        <li
          class="c16"
          data-testid="portfolio-page__list__item"
          style="animation-delay: 0.0375s;"
        >
          <a
            href="/portfolio/scrupulum-misphrase"
            role="listitem"
          >
            <figure
              class="c17"
            >
              <picture>
                <source
                  sizes="360px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=360 360w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=720 720w"
                  type="image/webp"
                />
                <source
                  sizes="360px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=360 360w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=720 720w"
                  type="image/jpeg"
                />
                <source
                  sizes="600px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=600 600w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=1200 1200w"
                  type="image/webp"
                />
                <source
                  sizes="600px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=600 600w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=1200 1200w"
                  type="image/jpeg"
                />
                <img
                  alt=""
                  class="c18"
                  data-testid="loadable-image"
                  loading="lazy"
                  src="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform"
                />
              </picture>
            </figure>
            <div
              class="c19"
            >
              <h3
                class="c20"
              >
                Scrupulum Misphrase
              </h3>
              <p
                class="c21"
              >
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo velit omnis.
              </p>
            </div>
          </a>
        </li>
        <li
          class="c16"
          data-testid="portfolio-page__list__item"
          style="animation-delay: 0.075s;"
        >
          <a
            href="/portfolio/scrupulum-misphrase"
            role="listitem"
          >
            <figure
              class="c17"
            >
              <picture>
                <source
                  sizes="360px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=360 360w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=720 720w"
                  type="image/webp"
                />
                <source
                  sizes="360px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=360 360w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=720 720w"
                  type="image/jpeg"
                />
                <source
                  sizes="600px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=600 600w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=webp&w=1200 1200w"
                  type="image/webp"
                />
                <source
                  sizes="600px"
                  srcset="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=600 600w, https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform?fm=jpg&w=1200 1200w"
                  type="image/jpeg"
                />
                <img
                  alt=""
                  class="c18"
                  data-testid="loadable-image"
                  loading="lazy"
                  src="https://endocrinopathic.com/pluviometrically/phosphorylase?a=ulnocarpal&b=proreptilia#trochleariform"
                />
              </picture>
            </figure>
            <div
              class="c19"
            >
              <h3
                class="c20"
              >
                Scrupulum Misphrase
              </h3>
              <p
                class="c21"
              >
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo velit omnis.
              </p>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
